graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/KotlinDataTypeGenerator.kt

```diff
@@ -181,6 +183,21 @@ abstract class AbstractKotlinDataTypeGenerator(packageName: String, protected va
                     else -> if (field.nullable) parameterSpec.defaultValue("null")
                 }
             }
+            funConstructorBuilder.addParameter(parameterSpec.build())
+        }
+        kotlinType.primaryConstructor(funConstructorBuilder.build())
```

Comment:
We want to make the _Kotlin Data Type_ to express the constructor attributes as type parameters, for example, have ```
Car(val foo: String = null)
```
or 
```
Car(var foo: String = null)
```

If the `generateAllConstructor` is disabled, this will allow us to _construct_ a _Kotlin Data Type_ by calling their no _argument constructor_ and then using the setter method